thrust/testing/transform_scan.cu

```diff
@@ -353,6 +426,22 @@ void TestValueCategoryDeduction()
   ASSERT_EQUAL(T{8}, vec[8]);
   ASSERT_EQUAL(T{9}, vec[9]);
 
+  vec.assign((T*) a_h, a_h + 10);
+
+  thrust::transform_inclusive_scan(
+    thrust::device, vec.cbegin(), vec.cend(), vec.begin(), thrust::identity<>{}, T{}, thrust::maximum<>{});
+
+  ASSERT_EQUAL(T{5}, vec[0]);
+  ASSERT_EQUAL(T{5}, vec[1]);
+  ASSERT_EQUAL(T{5}, vec[2]);
+  ASSERT_EQUAL(T{8}, vec[3]);
+  ASSERT_EQUAL(T{8}, vec[4]);
+  ASSERT_EQUAL(T{8}, vec[5]);
+  ASSERT_EQUAL(T{8}, vec[6]);
+  ASSERT_EQUAL(T{8}, vec[7]);
+  ASSERT_EQUAL(T{8}, vec[8]);
+  ASSERT_EQUAL(T{9}, vec[9]);
```

Comment:
```suggestion
  ASSERT_EQUAL(thrust::device_vector<T>{5, 5, 8, 8, 8, 8, 8, 8, 9}, vec);
```

Reply:
that's the spirit

Reply:
@bernhardmgruber I guess you missed a `5` here (?)

Reply:
I just checked whether you are paying attention. You passed ;)